Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -1576,9 +1577,21 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
         
         // TODO: When the symbol graph includes the precise identifiers for conditional availability, those symbols should also be resolved (rdar://63768609).
         
+        func resolveSymbol(symbolID: String) -> (ResolvedTopicReference, LinkResolver.ExternalEntity)? {
+            if let globalResult = globalExternalSymbolResolver?.symbolReferenceAndEntity(withPreciseIdentifier: symbolID) {
+                return globalResult
+            }
+            for externalResolver in linkResolver.externalResolvers.values {
+                if let result = externalResolver.symbolReferenceAndEntity(symbolID: symbolID) {
+                    return result
```

Comment:
I ran a test using your instructions but I wasn't able to get a link from one archive to another to work.

I setup a simple Swift package that used Swift Algorithms as a dependency. In my package, I wrote this code:

```
/// This is a class that can run something.
class MyClass {
    
    var chunks: ChunksOfCountCollection<Array<String>>
    
    init(names: [String]) {
        self.chunks = names.chunks(ofCount: 3)
    }
```

Then I built the documentation using the `--dependency` (passing the docc archive for Swift Algorithms) and `--enable-experimental-external-link-support` and was able to see the declaration for the `chunks` instance property. However, the link from this property's declaration to the underlying `ChunksOfCountCollection` type didn't work.

Running in the debugger, the `result` here contained this URL: `doc://Documentation/documentation/Algorithms/ChunksOfCountCollection` - it has a `doc:` protocol and not a normal path string.

Inside of ExternalPathHierarchyResolver, I did see the proper URL in the `relativePresentationURL` attribute of `resolvedInformation` inside of ExternalPathHierarchyResolver.entity:

```
relativePresentationURL "/documentation/algorithms/chunksofcountcollection"	
```
...but this wasn't used by the Render JSON and DocC Render of course couldn't render the link properly.

Am I doing something wrong? I did combine the two .doccarchives under a single web server, and I'm not seeing 404 errors.

Reply:
Nice catch. The entity was using the wrong value for its render reference's URL. I fixed this here https://github.com/apple/swift-docc/pull/826/commits/b849733e8a321941b2ddb88ec8748ba55b86e358.

While testing this end-to-end I also made additional auxiliary changes in https://github.com/apple/swift-docc/pull/826/commits/2c22bffbc7612fc41da48d17154dc96aff3583c5 and https://github.com/apple/swift-docc/pull/826/commits/f714ae6859eb9fbd5a235ba3e06b02a793d95512 that aren't about symbol reference in the symbol graph but that fix 2 bugs in the broader combined documentation experience.